src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPServerSideOnTypeFormattingEnterHandler.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.intellij.codeInsight.editorActions.enter.EnterHandlerDelegateAdapter;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Performs LSP-based on-type formatting if appropriate for the file when Enter is typed. For handling of other typed
+ * characters, see {@link LSPServerSideOnTypeFormattingTypedHandler}.
+ */
+public class LSPServerSideOnTypeFormattingEnterHandler extends EnterHandlerDelegateAdapter {
```

Comment:
This is the `enterHandlerDelegate` that just immediately invokes `LSPServerSideOnTypeFormattingHelper.applyOnTypeFormatting()` with a typed character of `\n`. If text edits were applied, it short-circuits any further `enterHandlerDelegate` implementations.